From a22c9b5dbcc084945d61692a1b98f8e3bcd3ad6d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 27 Jul 2021 13:13:31 +0300
Subject: [PATCH] [SW-50527] gaudi: security: remove tpc cfg/qm prot

The DBG TPC_CFG and DBG TPC QMAN pages are an exact mirror of the same
pages in the TPC CFG space, so we should not try to protect them twice.
This can result in exception because when we try to protect the second
time from the DBG space, we do not have the privilege level bit and try
to access the protected area.
Remove the above pages redundant protection.

Change-Id: I4681dd11dd8c74ae7bf47d6f744dbcb97836de3e
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 707cabc..c6b05c9 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1028,18 +1028,6 @@
 		 * initiator) will result in exception.
 		 */
 		regmap_writel(0, prot_base + 0xEF4);
-
-		/* Protect DBG TPC_CFG - only Priv protection and int bits */
-		prot_base = (mmTPC0_EML_TPC_CFG_BASE & ~HL_CFG_BASE) + base;
-		regmap_writel(0, prot_base + 0xEF8);
-		/* same reason as above */
-		regmap_writel(0, prot_base + 0xEF4);
-
-		/* Protect DBG TPC QMAN - only Priv protection and int bits */
-		prot_base = (mmTPC0_EML_TPC_QM_BASE & ~HL_CFG_BASE) + base;
-		regmap_writel(0, prot_base + 0xEF8);
-		/* same reason as above */
-		regmap_writel(0, prot_base + 0xEF4);
 	}
 }
 
